### PR TITLE
fix new chat w/ reset participants

### PR DESCRIPTION
**Pull request by chrisnojima**
Fixes a bug, heres the repro

1. Make an account on the web (`new`)
1. Add a pgp key
1. With another account (`other`) share some files in kbfs (`/keybase/private/new,other/hello.txt`)
1. Reset the `new` account
1. Start a chat convo `new,other`
1. You should see the red banner w/ skull and crossbones
1. Click `add them` button
1. Send to new chat

This should fail with a blackbar no tlfname specific error

### CODE REVIEW

**Comment by mmaxim** [comment]
What is the source of this inbox refresh?

**Comment by chrisnojima** [comment]
- [ ] understand where that refresh is coming from
- [ ] bookkeep these new convos and inject them on inbox refreshes

File: shared/actions/chat2/index.js [hunk 1]
```diff
@@ -165,9 +165,7 @@ const unboxRows = (
 
   const onUnboxed = function*({conv}: RPCChatTypes.ChatUiChatInboxConversationRpcParam) {
     const inboxUIItem: RPCChatTypes.InboxUIItem = JSON.parse(conv)
-    // We allow empty conversations now since we create them and they're empty now
-    const allowEmpty = action.type === Chat2Gen.selectConversation
-    const meta = Constants.inboxUIItemToConversationMeta(inboxUIItem, allowEmpty)
+    const meta = Constants.inboxUIItemToConversationMeta(inboxUIItem, true)
```
Comment (chrisnojima) [inline shared/actions/chat2/index.js]: always allow empty

File: shared/chat/conversation/messages/special-bottom-message.js [hunk 2]
```diff
@@ -46,6 +47,7 @@ const mapStateToProps = (state: TypedState, ownProps: OwnProps) => {
       : null
 
   return {
+    measure: ownProps.measure,
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/special-bottom-message.js]: this missing plumbing was what caused the red skull box to overlap sometimes (and in this case never appear!)

File: shared/util/saga.js [hunk 3]
```diff
@@ -143,7 +143,13 @@ function safeTakeEveryPure<A, R, FinalAction, FinalActionError>(
       }
 
       if (actionCreatorsWithResult) {
-        yield actionCreatorsWithResult(result, action)
+        if (actionCreatorsWithResult.length === 3) {
```
Comment (chrisnojima) [inline shared/util/saga.js]: special version that gets an updated state in the pure function

**Comment by chrisnojima** [comment]
@keybase/react-hackers cc: @mmaxim

**Pull request merged**